extensions/amp-form/amp-form.md

```diff
@@ -75,7 +75,108 @@ __required__
 Action must be provided, `https` and is non-cdn link (does **NOT** link to https://cdn.ampproject.org).
 
+**action-xhr**
+__optional__
+You can also provide an action-xhr attribute, if provided, the form will be submitted in an XHR fashion.
+
+This attribute can be the same or a different endpoint that action and has the same action requirement above.
+
+**Important**: Your XHR endpoints need to follow and implement [CORS Requests in AMP spec](https://github.com/ampproject/amphtml/blob/master/spec/amp-cors-requests.md).
+
 All other [form attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) are optional.
 
 ## Inputs
 Currently, `<input type=button>`, `<input type=file>`, `<input type=image>` and `<input type=password>` are not allowed. (This might be reconsidered in the future - please let us know if you require these and use cases).
+
+## Events
+`amp-form` exposes 3 events:
+**submit**
+Emitted whenever the form is submitted and before the submission is complete.
+
+**submit-success**
+Emitted whenever the form submission is done and response is a success.
+
+**submit-error**
+Emitted whenever the form submission is done and response is an error.
+
+These events can be used through the [on attribute](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-format.md#on).
+For example, the following listens to both submit success and submit error and shows different lightboxes depending on the event
+
+```html
+<form ... on="submit-success:success-lightbox;submit-error:error-lightbox" ...>
+</form>
+```
+
+See the [full example here](https://github.com/ampproject/amphtml/blob/master/examples/forms.amp.html).
+
+## Success/Error Response Rendering
+`amp-form` allows publishers to render the responses using [Extended Templates](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-format.md#extended-templates).
+
+Using `submit-success` and `submit-error` special marker attributes, publishers can mark any **child element of form** and include a `<template></template>` tag inside it to render the response in it.
+
+The response is expected to be a valid JSON Object. For example, if the publisher's `action-xhr` endpoint returns the following responses:
+
+**Success Response**
+```json
+{
+  "name": "Jane Miller",
+  "interests": [{"name": "Basketball"}, {"name": "Swimming"}, {"name": "Reading"}],
+  "email": "email@example.com"
+}
+```
+
+**Error Response**
+```json
+{
+  "name": "Jane Miller",
+  "message": "The email (email@example.com) you used is already subscribed."
+}
+```
+
+Publishers can render these in a template inside their forms as follow.
+
+```html
+<form ...>
+    <fieldset>
+      ...
+    </fieldset>
+    <div submit-success>
+        <template type="amp-mustache">
+            Success! Thanks {{name}} for subscribing! Please make sure to check your email {{email}}
+            to confirm! After that we'll start sending you weekly articles on {{#interests}}<b>{{name}}</b> {{{{/interests}}.
```

Comment:
can you add a new line here so that the there are no horizontal scrollbars on the example to see the full code